### PR TITLE
feat(ios): add resolving Apple based app-groups directory

### PR DESCRIPTION
**Pull request by MegaMaddin**
Hello,

I'm currently in the process of getting rid of rn-fetch-blob due to the fact that is is being [unmaintained](https://github.com/joltup/rn-fetch-blob/issues/666) for a while now. I stumbled across `rn-file-access` and like the simplicity and usage of modern languages.

To be able to switch over I need some additional features, starting with being able to resolve Apple's app-group directory for sharing content within an app-group. Please feel free to accept this pull-request.

## Testing done
Since app-groups require a proper entitlement including a signing configuration, I haven't checked in that code in the example app, but will provide a code snippet for doing this.

To be able to test this change, follow those steps

1. add the app-group entitlement in XCode and provide `group.FileAccessExample` as the group name
2. select a proper signing profile to be able to use app-groups
3. add the following code somewhere in `example/src/App.tsx` within the `setInfo` `useEffect`

```js
    // AppGroups on Apple devices
    FileSystem.getAppGroupDir('group.FileAccessExample').then((groupDir) => {
      setInfo((prev) => {
        prev.push({
          key: 'getAppGroupDir(group.FileAccessExample)',
          value: groupDir,
        });

        return prev.slice();
      });
    });
```
 4. this should provide the following content
```txt
getAppGroupDir(group.FileAccessExample) /Users/username/Library/Developer/CoreSimulator/Devices/DEADC0DE-BEEF-C0F3-C0FE-DEADBEEF/data/Containers/Shared/AppGroup/DEADC0DE-BEEF-C0F3-C0FE-DEADBEEF
```

Thanks,
Martin

### CODE REVIEW

File: ios/FileAccess.swift [hunk 1]
```diff
@@ -179,6 +179,18 @@ class FileAccess: NSObject {
...
+    @objc(getAppGroupDir:withResolver:withRejecter:)
+    func getAppGroupDir(groupName: String, resolve: RCTPromiseResolveBlock, reject: RCTPromiseRejectBlock) -> Void {
+        let groupURL = FileManager.default.containerURL(forSecurityApplicationGroupIdentifier: groupName)
+
+        if (groupURL == nil) {
+            reject("ERR", "Could not resolve app group directory. The group name '\(groupName)' is invalid.", nil)
+            return
+        }
+
+        resolve(groupURL?.path)
```
Comment (alpha0010) [inline ios/FileAccess.swift:191]: Could you use the `guard let` assignment to unrap the `nil`?  (See other examples in this file, or search swift docs for the `guard` statement.)

File: src/index.ts [hunk 2]
```diff
@@ -92,6 +92,18 @@ export const FileSystem = {
...
   },
 
+  /**
+   * Return the local storage directory for app groups.
+   *
+   * This is an Apple only feature.
+   */
+  getAppGroupDir(groupName: string) {
+    if (Platform.OS !== 'ios' && Platform.OS !== 'macos') {
+      return Promise.reject('AppGroups are available on Apple devices only');
```
Comment (alpha0010) [inline src/index.ts:102]: Please wrap string in an `Error` object.

**Review by alpha0010 — CHANGES_REQUESTED** [review]
Thanks for your pr; a couple changes and it should be good to go.

File: ios/FileAccess.swift (hunk 1 above)
Comment (MegaMaddin) [reply ios/FileAccess.swift:191]: Sure, I can do that!

File: src/index.ts (hunk 2 above)
Comment (MegaMaddin) [reply src/index.ts:102]: Yep, will do.

**Review by alpha0010 — APPROVED** [review]

**Pull request merged**